src/main/kotlin/uk/gov/justice/digital/hmpps/manageoffencesapi/service/OffenceService.kt

```diff
@@ -23,6 +30,125 @@ class OffenceService(
+    ('A'..'Z').forEach { alphaChar ->
+      SDRSService.log.info("Starting full sync with NOMIS for alphaChar {} ", alphaChar)
+      fullySyncOffenceGroupWithNomis(alphaChar.toString())
+    }
+  }
+
+  private fun fullySyncOffenceGroupWithNomis(alphaChar: String) {
+    val offencesByCode = offenceRepository.findByCodeStartsWithIgnoreCase(alphaChar).associateBy { it.code }
+    val (nomisOffencesById, nomisOffences) = getAllNomisOffencecsForAlphaChar(alphaChar)
+
+    // the keys here represent the NOMIS keys, each key is a pair of the offenceCode and the statuteCode
+    val (existingOffenceKeys, newOffenceKeys) = offencesByCode.keys.map { Pair(it, it.substring(0, 4)) }
+      .partition { nomisOffencesById.containsKey(it) }
+
+    newOffenceKeys.forEach {
+      val offence = offencesByCode[it.first]!!
+      val statute = createStatuteInNomis(offence, nomisOffences)
+      val homeOfficeCode = createHomeOfficeCodeInNomis(offence, nomisOffences)
+      createOffenceInNomis(offence, statute, homeOfficeCode)
+    }
+
+    existingOffenceKeys.forEach {
+      val offence = offencesByCode[it.first]!!
+      val nomisOffence = nomisOffencesById[it]!!
+      val homeOfficeCode = createHomeOfficeCodeInNomis(offence, nomisOffences)
+      updateOffenceInNomis(offence, nomisOffence, homeOfficeCode)
+    }
+  }
+
+  private fun updateOffenceInNomis(
+    offence: EntityOffence,
+    nomisOffence: PrisonApiOffence,
+    homeOfficeCode: PrisonApiHoCode?
+  ) {
+    val nomisOffenceUpdated = nomisOffence.copy(
+      description = offence.description,
+      hoCode = homeOfficeCode,
+      // severityRanking = "99", // TODO should derive this and how to set on update?
+      activeFlag = isOffenceActive(offence.endDate),
+    )
+    prisonApiClient.updateOffence(nomisOffenceUpdated)
+  }
+
+  private fun createOffenceInNomis(
+    offence: EntityOffence,
+    statute: PrisonApiStatute,
+    homeOfficeCode: PrisonApiHoCode?
+  ) {
+    val nomisOffence = PrisonApiOffence(
+      code = offence.code,
+      description = offence.description,
+      statuteCode = statute,
+      hoCode = homeOfficeCode,
+      severityRanking = "99", // TODO should derive this?
+      activeFlag = isOffenceActive(offence.endDate),
+    )
+    prisonApiClient.createOffence(nomisOffence)
+  }
+
+  private fun isOffenceActive(endDate: LocalDate?): String {
+    if (endDate == null || endDate.isAfter(LocalDate.now())) return "Y"
+    return "N"
+  }
+
+  private fun createHomeOfficeCodeInNomis(
+    offence: EntityOffence,
+    nomisOffences: List<PrisonApiOffence>
+  ): PrisonApiHoCode? {
+    // TODO Make sure format of hoCode is 6 chars both sides
+    if (offence.homeOfficeStatsCode == null) return null
+    // TODO replace this condition, get all ho-codes from prison-api (requires new endpoint) and only create one if the statute doesnt exist in the list
+    if (nomisOffences.any { it.hoCode != null && it.hoCode.code == offence.homeOfficeStatsCode }) {
```

Comment:
Not sure but I think you can do the null check by `it.hoCode?.equals(offence.homeOfficeStatsCode)`

Reply:
Thanks Mike, good spot will change. 
FYI this gonna change this as well, this is an interim solution until we create a end point in prisonAPI to fetch all ho codes